`nix store ping` -> `nix store info`

vicky1999 opened the pull request:
# Motivation
This change will create alias of `store info`.  

# Context
This fix is part of #8914 
In `ping-store.cc`, I have registered another command for CmdPingStore for store info which will register the command and can be used with store commands.

Implemented as per the discussion in #8975 PR

### Demo:
![image](https://github.com/NixOS/nix/assets/24916780/259fe575-71f5-48a7-8d6f-fda3f1c62526)

cc: @Ericson2314 @fricklerhandwerk 

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

fricklerhandwerk commented:
Hey, #8975 looked like it was far closer to implementing the issue. Here the new command name is the one that issues a warning, but it should be the old one acting as an alias. What about resurrecting the other PR?

vicky1999 commented:
@fricklerhandwerk I had few issues with my forked repo and it made harder to fix the new changes.  So, I have closed all the PRs and re-forked the repo again and implemented the changes as per the suggestions.  Because of that the old PRs were closed.

Ericson2314 commented:
I did suggest @vicky1999 do it this way as a stop-gap based on prior art, so I think this is a fine code change. But yes we want the existing use-cases of `nix store ping` to be rewritten to be `nix store info` e.g. in the tests.

vicky1999 commented:
@Ericson2314 If you want, I can change the alias to be ping and main command to be info just by changing the `registerCommand2` function

vicky1999 commented:
@fricklerhandwerk  @Ericson2314 
I have updated the help content and also changed the alias to be `nix store ping` instead of `nix store info`.  Kindly check this screenshot below and let me know if any changes were needed.

![image](https://github.com/NixOS/nix/assets/24916780/1ab529cc-b19d-4021-8dd1-6adbbf3a4d2a)

Ericson2314 commented:
This looks good, thanks @vicky1999!

vicky1999 commented:
@Ericson2314 I have renamed the file in store ping to store info in other sh files and also updated the new filename in `tests/functional/local.mk`

fricklerhandwerk commented:
@vicky1999 thanks for keeping at it! Good job. @Ericson2314 thanks for taking the time to help with guidance.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/2023-10-16-nix-team-meeting-minutes-95/34396/1

iFreilicht commented:
@Ericson2314 @fricklerhandwerk Wasn't there a decision to not modify release notes of past releases?

fricklerhandwerk commented:
Oh yeah that was a slip-up. Feel free to PR it away.
